R/T_and_F_symbol_linter.R

```diff
@@ -4,28 +4,55 @@
 #' @export
 T_and_F_symbol_linter <- function() { # nolint: object_name_linter.
   Linter(function(source_file) {
-    lapply(
-      ids_with_token(source_file, "SYMBOL"),
-      function(id) {
-        token <- with_id(source_file, id)
-        symbol <- re_matches(token[["text"]], rex(start, capture(or("T", "F")), end))[1L, 1L]
-        if (!is.na(symbol)) {
-          replacement <- switch(symbol, "T" = "TRUE", "F" = "FALSE")
-          line_num <- token[["line2"]]
-          start_col_num <- token[["col1"]]
-          end_col_num <- token[["col2"]]
-          Lint(
-            filename = source_file[["filename"]],
-            line_number = line_num,
-            column_number = end_col_num + 1L,
-            type = "style",
-            message = sprintf("Use %s instead of the symbol %s.", replacement, symbol),
-            line = source_file[["lines"]][[as.character(line_num)]],
-            ranges = list(c(start_col_num, end_col_num)),
-            linter = "T_and_F_symbol_linter"
-          )
-        }
-      }
+    if (is.null(source_file$xml_parsed_content)) return(list())
+
+    xpath <- paste0(
+      "//SYMBOL[",
+      "(text() = 'T' or text() = 'F')", # T or F symbol
+      " and count(preceding-sibling::OP-DOLLAR) = 0", # not part of a $-subset expression
```

Comment:
`not(preceding-sibling::OP-DOLLAR)` is more "canonical XPath" to me (not that I have any real expertise to call XPath canonical :sweat_smile:)

Reply:
Sounds more idiomatic than `count() = 0` 👍 
I'll change this later.

Reply:
Done